src/lib.rs

```diff
@@ -83,12 +105,15 @@ pub fn run(args: Args) {
             .expect(&format!("Unknown capture name: `{}`", capture_name))
     });
     let output_mode = get_output_mode(&args);
+    let buffer_writer = BufferWriter::stdout(ColorChoice::Never);
```

Comment:
Ok this stuff is for avoiding the interleaving printed matches (which I started seeing consistently once I added another searched file to the `rust_project/` test fixture directory)

[This method](https://github.com/BurntSushi/ripgrep/blob/master/crates/core/main.rs#L127) is where you can see how `ripgrep` is doing it, and I just mimicked what seemed to be the relevant parts (with some changes to account for the fact that we're doing separate multi-threading for running the queries (via `rayon`) vs they are just using the `WalkParallel.run()` callback for multi-threading)

Specifically here a `BufferWriter` I believe is responsible for avoiding interleaving

And so we spawn/cache a printer + searcher per thread and the printer leverages the `BufferWriter` or something like that